dragonfly/implementations/daq_run_interface.py

```diff
@@ -108,18 +125,34 @@ def start_run(self, run_name):
         self.run_name = run_name
         self._run_meta = {'DAQ': self.daq_name,
                          }
+        self._prerun_snapshot = {}
         self._do_prerun_gets()
         if not self._debug_without_meta_broadcast:
             self._send_metadata()
+        if not self._debug_without_snapshot_broadcast:
+            self._send_snapshot(snap_flag='pre')
         logger.debug('these meta will be {}'.format(self._run_meta))
         logger.info('start_run finished')
 
     def _do_prerun_gets(self):
-        logger.info('doing prerun meta-data gets')
-        result = self.provider.get(self._metadata_state_target, timeout=120)
-        these_metadata = result['value_raw']
+        logger.info('doing prerun meta-data and snapshot gets')
+        meta_result = self.provider.get(self._metadata_state_target, timeout=120)
+        these_metadata = meta_result['value_raw']
         self._run_meta.update(these_metadata)
-        self.determine_RF_ROI()
+        for target,item_list in self._snapshot_target_items.items():
+            snapshot_result = self.provider.cmd(target, 'get_latest', [self._start_time,item_list], timeout=120)
+            these_snaps = snapshot_result['value_raw']
+            self._prerun_snapshot.update(these_snaps)
+        if not self._debug_without_rf_roi:
```

Comment:
The RF_ROI goes into the metadata file which becomes tag values in DIRAC. That file gets sent to the metadata receiver prior to the start of data taking and may (@nsoblath ?) be responsible for triggering making a directory for the run. Dirac needs this info in order to apply the tags to the data when they are added to the catalog.

Line 155 is data for the post-run snapshot, which goes into context files which help interpret the data but are not dirac tags and which do include logged values over the course of the run (so the must be after).